File: Data Structures & Algorithms/longest-common-prefix/submission-2.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        res = ''
        for i in range(len(strs[0])):
            for char in strs:
                if i == len(char) or char[i] != strs[0][i]:
                    return res

            res += strs[0][i]
        
        return res

File: Data Structures & Algorithms/subsets/submission-2.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<int> path;
        vector<vector<int>> res;

        dfs(0, nums, path, res);
        return res;
    }

    void dfs(int c, vector<int>&nums, vector<int> p, vector<vector<int>>& r)
    {
        if(c == nums.size())
        {
            r.push_back(p);
            return;
        }

        
        dfs(c + 1, nums, p, r);

        p.push_back(nums[c]);
        dfs(c + 1, nums, p, r);
        p.pop_back();
    }
};
